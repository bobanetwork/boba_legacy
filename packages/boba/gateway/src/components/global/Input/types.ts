import { ReactNode } from 'react'

export interface InputProps {
  placeholder: string
  type: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: any
  label?: ReactNode
}
