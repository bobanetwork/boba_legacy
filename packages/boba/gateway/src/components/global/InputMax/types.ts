export interface MaxInputTypes {
  max: number
  initialValue?: number
  onValueChange: (value: number) => void
}
