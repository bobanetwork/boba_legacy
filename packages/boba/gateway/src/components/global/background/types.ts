export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
