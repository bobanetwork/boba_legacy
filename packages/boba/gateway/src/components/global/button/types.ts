import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
