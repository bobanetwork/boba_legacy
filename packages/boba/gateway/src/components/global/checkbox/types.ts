export interface CheckboxWithLabelProps {
  label?: string
  checked?: boolean
  onChange: (isChecked: boolean) => void
}
