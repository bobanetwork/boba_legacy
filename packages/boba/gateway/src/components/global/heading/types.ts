import { ReactNode } from 'react'

export type VariantType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

export interface HeadingProps {
  variant: VariantType
  color?: string
  children?: ReactNode
  className?: string
}

export interface HeadingStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
