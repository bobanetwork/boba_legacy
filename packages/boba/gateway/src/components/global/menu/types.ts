import { ComponentType, ReactElement, ReactNode } from 'react'

export interface OptionProps {
  label?: string
  component?: ReactElement<any>
  onClick?: (d: any) => void
}

export interface MenuProps {
  name?: string
  label?: ReactNode | string
  children?: ReactNode
  options: Array<OptionProps>
}
