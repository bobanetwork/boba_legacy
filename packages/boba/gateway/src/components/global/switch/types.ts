export interface SwitchButtonTypes {
  isDisable?: boolean
  isActive?: boolean
  onStateChange?: (isChecked: boolean) => void
}
