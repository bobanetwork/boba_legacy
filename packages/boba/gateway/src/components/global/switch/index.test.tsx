import React from 'react'
import { render } from '@testing-library/react'
import { SwitchButtonTypes } from './types'
import CustomThemeProvider from 'themes'
import { SwitchButton } from './'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const renderSwitch = (props: SwitchButtonTypes) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'light',
        },
      })}
    >
      <CustomThemeProvider>
        <SwitchButton {...props} />
      </CustomThemeProvider>
    </Provider>
  )
}

describe('SwitchButton', () => {
  test('SwitchButton Default State', () => {
    const { asFragment } = renderSwitch({
      onStateChange: jest.fn(),
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('SwitchButton is Active', () => {
    const { asFragment } = renderSwitch({
      isActive: true,
      onStateChange: jest.fn(),
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('SwitchButton is Disable', () => {
    const { asFragment } = renderSwitch({
      isDisable: true,
      onStateChange: jest.fn(),
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('SwitchButton is Disable & Active', () => {
    const { asFragment } = renderSwitch({
      isActive: true,
      isDisable: true,
      onStateChange: jest.fn(),
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
