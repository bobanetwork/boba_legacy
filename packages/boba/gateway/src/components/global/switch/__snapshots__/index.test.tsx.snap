// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwitchButton SwitchButton Default State 1`] = `
<DocumentFragment>
  <label
    class="sc-ewnqHT byTfLr"
  >
    <input
      class="sc-gTRrQi fYYSjw"
      type="checkbox"
    />
    <span
      class="sc-bgqQcB fNGpeI"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton SwitchButton is Active 1`] = `
<DocumentFragment>
  <label
    class="sc-ewnqHT byTfLr"
  >
    <input
      checked=""
      class="sc-gTRrQi fYYSjw"
      type="checkbox"
    />
    <span
      class="sc-bgqQcB fNGpeI"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton SwitchButton is Disable & Active 1`] = `
<DocumentFragment>
  <label
    class="sc-ewnqHT byTfLr"
  >
    <input
      checked=""
      class="sc-gTRrQi fYYSjw"
      disabled=""
      type="checkbox"
    />
    <span
      class="sc-bgqQcB fNGpeI"
    />
  </label>
</DocumentFragment>
`;

exports[`SwitchButton SwitchButton is Disable 1`] = `
<DocumentFragment>
  <label
    class="sc-ewnqHT byTfLr"
  >
    <input
      class="sc-gTRrQi fYYSjw"
      disabled=""
      type="checkbox"
    />
    <span
      class="sc-bgqQcB fNGpeI"
    />
  </label>
</DocumentFragment>
`;
